code/crates/core-consensus/src/handle/decide.rs

```diff
@@ -1,25 +1,67 @@
 use crate::prelude::*;
 
 #[cfg_attr(not(feature = "metrics"), allow(unused_variables))]
-pub async fn decide<Ctx>(
+pub async fn try_decide<Ctx>(
     co: &Co<Ctx>,
     state: &mut State<Ctx>,
     metrics: &Metrics,
-    consensus_round: Round,
-    proposal: SignedProposal<Ctx>,
 ) -> Result<(), Error<Ctx>>
 where
     Ctx: Context,
 {
-    let height = proposal.height();
-    let proposal_round = proposal.round();
-    let value = proposal.value();
+    if !state.driver.step_is_commit() {
+        return Ok(());
```

Comment:
Let's log something here, as this is not supposed to happen during normal operations, right?

Reply:
it can happen when we sync or WAL replay

Reply:
Right, then it's fine.